whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NrtmFileRepository.java

```diff
@@ -36,32 +36,29 @@ public class NrtmFileRepository {
     private final DateTimeProvider dateTimeProvider;
     private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
 
-    public NrtmFileRepository(@Qualifier("nrtmDataSource") final DataSource dataSource, final DateTimeProvider dateTimeProvider) {
+
+    public NrtmFileRepository(@Qualifier("nrtmSourceAwareDataSource") final DataSource dataSource, final DateTimeProvider dateTimeProvider) {
         this.jdbcTemplate = new JdbcTemplate(dataSource);
         this.dateTimeProvider = dateTimeProvider;
         this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(jdbcTemplate);
     }
 
-    @Transactional
+    @Transactional("nrtmTransactionManager")
     public void saveDeltaVersion(final NrtmVersionInfo version, final int serialIDTo, final List<DeltaFileRecord> deltas) throws JsonProcessingException {
```

Comment:
<bean id="nrtmTransactionManager" class="org.springframework.jdbc.datasource.DataSourceTransactionManager">
        <property name="dataSource" ref="nrtmDataSource"/>
    </bean> 

NrtmTransaction uses nrtmDataSource (which is nrtm master datasource , we should rename it to be consistent with whois naming convention ) why we need to use sourceAware for trasactions? Is it being overriden?

Reply:
we don't need it, but Ed told me to keep there for being sure that it is using that transaction

Reply:
Using just master datasources and change the name to the source for nrtmMasterDataSource